mmv1/third_party/terraform/services/resourcemanager/ephemeral_google_service_account_token_test.go

```diff
@@ -0,0 +1,419 @@
+				t.Errorf("expected error, got none")
+			}
+
+			if !test.expectError && response.Diagnostics.HasError() {
+				t.Errorf("got unexpected error: %s", response.Diagnostics.Errors())
+			}
+
+			if test.errorContains != "" {
+				foundError := false
+				for _, err := range response.Diagnostics.Errors() {
+					if err.Summary() == test.errorContains {
+						foundError = true
+						break
+					}
+				}
+				if !foundError {
+					t.Errorf("expected error with summary %q, got none", test.errorContains)
+				}
+			}
+		})
+	}
+}
+
+func TestServiceScopeValidator(t *testing.T) {
+	t.Parallel()
+
+	type testCase struct {
+		value       types.String
+		expectError bool
+	}
+
+	tests := map[string]testCase{
+		"canonical form": {
+			value:       types.StringValue("https://www.googleapis.com/auth/cloud-platform"),
+			expectError: false,
+		},
+		"non-canonical form": {
+			value:       types.StringValue("cloud-platform"),
+			expectError: false,
+		},
+		"empty string": {
+			value:       types.StringValue(""),
+			expectError: false,
+		},
+	}
+
+	for name, test := range tests {
+		name, test := name, test
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			request := validator.StringRequest{
+				Path:           path.Root("test"),
+				PathExpression: path.MatchRoot("test"),
+				ConfigValue:    test.value,
+			}
+			response := validator.StringResponse{}
+			validator := resourcemanager.ServiceScopeValidator{}
+
+			validator.ValidateString(context.Background(), request, &response)
+
+			if test.expectError && !response.Diagnostics.HasError() {
+				t.Errorf("expected error, got none")
+			}
+
+			if !test.expectError && response.Diagnostics.HasError() {
+				t.Errorf("got unexpected error: %s", response.Diagnostics.Errors())
+			}
+		})
+	}
+}
+
+func TestAccEphemeralServiceAccountToken_basic(t *testing.T) {
+	t.Parallel()
+
+	serviceAccount := envvar.GetTestServiceAccountFromEnv(t)
+	targetServiceAccountEmail := acctest.BootstrapServiceAccount(t, "basic", serviceAccount)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() { acctest.AccTestPreCheck(t) },
+		ExternalProviders: map[string]resource.ExternalProvider{
+			"time": {},
+		},
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccEphemeralServiceAccountToken_basic(targetServiceAccountEmail),
+			},
+		},
+	})
+}
+
+func TestAccEphemeralServiceAccountToken_withDelegates(t *testing.T) {
+	t.Parallel()
+
+	project := envvar.GetTestProjectFromEnv()
+	initialServiceAccount := envvar.GetTestServiceAccountFromEnv(t)
+	delegateServiceAccountEmailOne := acctest.BootstrapServiceAccount(t, "delegate1", initialServiceAccount)          // SA_2
+	delegateServiceAccountEmailTwo := acctest.BootstrapServiceAccount(t, "delegate2", delegateServiceAccountEmailOne) // SA_3
+	targetServiceAccountEmail := acctest.BootstrapServiceAccount(t, "target", delegateServiceAccountEmailTwo)         // SA_4
```

Comment:
The acctest.BootstrapServiceAccount function creates a service account (or gets if it exists) and [gives another service account permissions on the service account](https://github.com/hashicorp/terraform-provider-google/blob/45df35f12441f4d6749a76f35d26fd89647aa941/google/acctest/bootstrap_test_utils.go#L238). For example in this line:

```go
delegateServiceAccountEmailOne := acctest.BootstrapServiceAccount(t, "delegate1", initialServiceAccount)          // SA_2
```

You are making sure there's an SA called `tf-bootstrap-sa-delegate1` and [you're giving `initialServiceAccount` the role `roles/iam.serviceAccountTokenCreator` on that SA.](https://github.com/hashicorp/terraform-provider-google/blob/45df35f12441f4d6749a76f35d26fd89647aa941/google/acctest/bootstrap_test_utils.go#L196-L205)

Because of this I think you should remove the `google_service_account_iam_binding` resources from your acc test config; you've already set needed permissions by choosing to use acctest.BootstrapServiceAccount and your config will affect any other acctests that try to use the same bootstrapped service account as you.

Reply:
just made the changes. I must've not had credentials setup correctly when initially working on this. Tried this right now and worked with no issues.